packages/peregrine/lib/talons/CheckoutPage/ShippingInformation/useShippingInformation.js

```diff
@@ -0,0 +1,79 @@
+import { useCallback, useEffect, useMemo } from 'react';
+import { useLazyQuery } from '@apollo/react-hooks';
+
+import { useAppContext } from '../../../context/app';
+import { useCartContext } from '../../../context/cart';
+import { MOCKED_ADDRESS } from '../../CartPage/PriceAdjustments/ShippingMethods/useShippingForm';
+
+export const useShippingInformation = props => {
+    const {
+        onSave,
+        queries: { getShippingInformationQuery }
+    } = props;
+
+    const [, { toggleDrawer }] = useAppContext();
+    const [{ cartId }] = useCartContext();
+
+    const [fetchShippingInformation, { called, data, loading }] = useLazyQuery(
+        getShippingInformationQuery
+    );
+
+    useEffect(() => {
+        if (cartId) {
+            fetchShippingInformation({
+                variables: {
+                    cartId
+                }
+            });
+        }
+    }, [cartId, fetchShippingInformation]);
+
+    const shippingData = useMemo(() => {
+        let filteredData;
+        if (data) {
+            const { cart } = data;
+            const { email, shipping_addresses: shippingAddresses } = cart;
+            if (shippingAddresses.length) {
+                const primaryAddress = shippingAddresses[0];
+                for (const field in MOCKED_ADDRESS) {
+                    if (primaryAddress[field] === MOCKED_ADDRESS[field]) {
+                        primaryAddress[field] = '';
+                    }
+
+                    if (
+                        field === 'street' &&
+                        primaryAddress[field][0] === MOCKED_ADDRESS[field][0]
+                    ) {
+                        primaryAddress[field] = [''];
+                    }
+                }
+
+                filteredData = {
+                    email,
+                    ...primaryAddress
+                };
+            }
+        }
+
+        return filteredData;
+    }, [data]);
+
+    const doneEditing = !!shippingData && !!shippingData.city;
```

Comment:
This is fragile but works - a comment here and wherever else we use it explaining why we use `city` to indicate completeness would be nice. If someone customizes their cart form with a city field this would no longer work.

Reply:
Added in 14a23f7